src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -79,20 +79,20 @@ public override void Initialize(AnalysisContext context)
 
         private void AnalyzeNode(SyntaxNodeAnalysisContext context, INamedTypeSymbol formatProviderType)
         {
-            var optionSet = context.Options.GetDocumentOptionSetAsync(
-                    context.Node.SyntaxTree, context.CancellationToken).GetAwaiter().GetResult();
+            var syntaxFacts = GetSyntaxFactsService();
+            var expression = syntaxFacts.GetExpressionOfInvocationExpression(context.Node);
 
-            if (optionSet.GetOption(
-                    ValidateFormatStringOption.ReportInvalidPlaceholdersInStringDotFormatCalls,
-                    context.SemanticModel.Language) == false)
+            if (!IsValidFormatMethod(syntaxFacts, expression))
```

Comment:
Is it worth putting a comment that we're putting the cheaper test first?

Reply:
I would expect other changes to reduce the cost of getting the option set by the time anyone cared to look. In other words, I believe such a comment in code has a high risk of becoming incorrect over time.

Reply:
The problem with a lack of comment is that there's nothing stopping someone x months/years from now from saying "this is inconssitent with other features, let me add the option check up front."  Sometimes it's good to say when something is intentional so that you're not limited by institutional memory as to why things were done.  

Now, this was much more of an issue in the past (when code actually moved from slime, to SD, to TFS, to github).  Hopefully we'll be on github a long time, so people can actually look up history effectively and know what the reasoning was for things like this.  But a simple one-line comment is very cheap, and can provide a lot of insight to help prevent regressions in the future.

Reply:
I believe found a better way to document it than just a comment. 😄